### PR TITLE
Run prettier in individual workspaces

### PR DESCRIPTION
**Pull request by mehulkar**
This adds a `lint:prettier` task in each workspace so that we can get the benefits of a granular cache and when running prettier, we don't have to run it across the entire monorepo (notably the `benchmark/` workspace which is very slow to run prettier on, and rarely changes -- an ideal candidate for a cache hit)

Note: adding this granularity immediately calls attention to the fact that we need some way to audit which workspaces may have forgotten to implement the script.

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -9,11 +9,10 @@
     // lint tasks
     "lint": {},
     // root lint task runs toml check and prettier
-    // TODO: run prettier in individual workspaces instead of globally.
     "//#lint": {
-      "dependsOn": ["//#check:toml", "//#check:prettier"]
+      "dependsOn": ["//#check:toml", "lint:prettier"]
     },
-    "//#check:prettier": {},
+    "lint:prettier": {},
```
Comment (nathanhammond) [inline turbo.json:15]: We should be smarter about setting `inputs` here. Make it just the places that `prettier` is going to search:

***

Update:

`prettier --support-info | jq "[.languages[].extensions[]] | unique"`

`pnpm prettier --support-info | jq "[.languages[].filenames | select( . != null )[]] | unique"`


Wow that's a lot of places, we should make this automatic.

```
[
  ".arcconfig",
  ".auto-changelog",
  ".babelrc",
  ".c8rc",
  ".clang-format",
  ".clang-tidy",
  ".devcontainer.json",
  ".eslintrc",
  ".eslintrc.json",
  ".gemrc",
  ".htmlhintrc",
  ".imgbotconfig",
  ".jscsrc",
  ".jshintrc",
  ".jslintrc",
  ".lintstagedrc",
  ".nycrc",
  ".prettierrc",
  ".stylelintrc",
  ".swcrc",
  ".tern-config",
  ".tern-project",
  ".watchmanconfig",
  "CITATION.cff",
  "Jakefile",
  "Pipfile.lock",
  "README",
  "api-extractor.json",
  "composer.json",
  "composer.lock",
  "contents.lr",
  "devcontainer.json",
  "glide.lock",
  "jsconfig.json",
  "language-configuration.json",
  "mcmod.info",
  "package-lock.json",
  "package.json",
  "tsconfig.json",
  "tslint.json"
]
```

```
[
  ".4DForm",
  ".4DProject",
  ".JSON-tmLanguage",
  "._js",
  ".avsc",
  ".bones",
  ".cjs",
  ".code-snippets",
  ".component.html",
  ".css",
  ".cts",
  ".es",
  ".es6",
  ".frag",
  ".geojson",
  ".gltf",
  ".gql",
  ".graphql",
  ".graphqls",
  ".gs",
  ".handlebars",
  ".har",
  ".hbs",
  ".hta",
  ".htm",
  ".html",
  ".html.hl",
  ".ice",
  ".importmap",
  ".inc",
  ".jake",
  ".javascript",
  ".js",
  ".js.flow",
  ".jsb",
  ".jscad",
  ".jsfl",
  ".jslib",
  ".jsm",
  ".json",
  ".json5",
  ".jsonc",
  ".jspre",
  ".jss",
  ".jsx",
  ".less",
  ".livemd",
  ".markdown",
  ".mcmeta",
  ".md",
  ".mdown",
  ".mdwn",
  ".mdx",
  ".mir",
  ".mjml",
  ".mjs",
  ".mkd",
  ".mkdn",
  ".mkdown",
  ".mts",
  ".njs",
  ".pac",
  ".pcss",
  ".postcss",
  ".reek",
  ".ronn",
  ".rviz",
  ".scd",
  ".scss",
  ".sjs",
  ".ssjs",
  ".sublime-build",
  ".sublime-commands",
  ".sublime-completions",
  ".sublime-keymap",
  ".sublime-macro",
  ".sublime-menu",
  ".sublime-mousemap",
  ".sublime-project",
  ".sublime-settings",
  ".sublime-syntax",
  ".sublime-theme",
  ".sublime-workspace",
  ".sublime_metrics",
  ".sublime_session",
  ".syntax",
  ".tfstate",
  ".tfstate.backup",
  ".topojson",
  ".ts",
  ".tsx",
  ".vue",
  ".webapp",
  ".webmanifest",
  ".workbook",
  ".wxs",
  ".wxss",
  ".xht",
  ".xhtml",
  ".xsjs",
  ".xsjslib",
  ".yaml",
  ".yaml-tmlanguage",
  ".yaml.sed",
  ".yml",
  ".yml.mysql",
  ".yy",
  ".yyp"
]
```

**Review by nathanhammond — CHANGES_REQUESTED** [review]

**Comment by tknickman** [comment]
@mehulkar checking in on this one - I can take another look once conflicts are resolved if we still want to get it in

**Review by tknickman — COMMENTED** [review]
@mehulkar checking in on this one - I can take another look once conflicts are resolved if we still want to get it in

**Comment by mehulkar** [comment]
@tknickman take a look, I think it should be good to go now

**Review by tknickman — COMMENTED** [review]
Looks like there's still an issue here with go lint:

```
make: golangci-lint: No such file or directory
```

File: turbo.json [hunk 2]
```diff
@@ -6,14 +6,170 @@
...
+    // root lint task runs toml check
     "//#lint": {
-      "dependsOn": ["//#check:toml", "//#check:prettier"]
+      "dependsOn": ["//#check:toml"]
     },
-    "//#check:prettier": {},
+
+    // run lint:prettier in all workspaces
+    "lint:prettier": {
+      "inputs": [
```
Comment (chris-olszewski) [inline turbo.json:21]: I'm guessing these are all the extensions that prettier will run against. In favor of us keeping our `turbo.json` more readable, I'd prefer if we just added the ones we use. In order to future proof from some bad cache hits we could add all of these patterns to `.prettierignore`.

Comment (mehulkar) [reply turbo.json:24]: you want to fight it out with @nathanhammond ?
https://github.com/vercel/turbo/pull/3925#discussion_r1115353284

**Comment by mehulkar** [comment]
The failing tests are because cli tests are running in the JS package workflow, when that workflow doesn't provision Go/Rust setup. This was reverted in 4e035e61eae802cd472dda40779e141d2e896659, so this failure would be fine if I updated the branch. But I don't want to wait for another round of CI checks if I do that.

**Pull request merged**